### PR TITLE
Fixes EUCA-12642

### PR DESCRIPTION
**Pull request by jbreiding**
loading the instance struct should not modify the instance xml and should assume any upgrades have already been performed.

### CODE REVIEW

File: storage/backing.c [hunk 1]
```diff
@@ -724,35 +724,6 @@ ncInstance *load_instance_struct(const char *instanceId)
...
-            }
-            EUCA_FREE(vol_data);
-        }
-    }
-
-    // save the struct back to disk after the upgrade routine had a chance to modify it
-    if (gen_instance_xml(instance) != EUCA_OK) {
-        LOGERROR("failed to create instance XML in %s\n", instance->xmlFilePath);
-        goto free;
-    }
```
Comment (euca-nightfury) [inline storage/backing.c]: Is it okay to remove the logic that saves the instance structure to the disk? It looks like there may be logic that modifies the cached instance prior to the volume update code.

Comment (euca-nightfury) [inline storage/backing.c]: I would recommend putting back the following logic
```
// save the struct back to disk after the upgrade routine had a chance to modify it
if (gen_instance_xml(instance) != EUCA_OK) {
    LOGERROR("failed to create instance XML in %s\n", instance->xmlFilePath);
    goto free;
}
```

Comment (jbreiding) [inline storage/backing.c]: That should only occur with the presence of the checkpoint file which shouldn't happen for a 4.3.1 and later upgrade right?

It's likely safe to remove that logic as well. But it actually does more than just "load instance struct" if it involves writing back the instance struct to disk. Perhaps post instance loading there could be an audit and reconcile phase that happens elsewhere and only when necessary.

Comment (jbreiding) [inline storage/backing.c]: I seem no harm in putting it back, so it is back.

**Comment by euca-nightfury** [comment]
Looks good to me. Let me know if you need any help reproducing the issue or testing the fix.

**Comment by jbreiding** [comment]
Squash merged to make for simpler cherry-picking.

**Pull request closed without merging**